webrtc.html

```diff
@@ -840,50 +840,48 @@ <h4><dfn>RTCPeerConnectionState</dfn> Enum</h4>
                 <th colspan="2">Enumeration description</th>
               </tr>
               <tr>
-                <td><dfn data-idl><code>new</code></dfn></td>
-                <td>Any of the <code><a>RTCIceTransport</a></code>s or
-                <code><a>RTCDtlsTransport</a></code>s are in the
-                <code>"new"</code> state and none of the transports are in the
-                <code>"connecting"</code>, <code>"checking"</code>,
-                <code>"failed"</code> or <code>"disconnected"</code> state, or all
-                transports are in the <code>"closed"</code> state, or there are
-                no transports.</td>
+                <td><dfn data-idl><code>closed</code></dfn></td>
+                <td>
+                  The <code><a>RTCPeerConnection</a></code> object's
+                  <a>[[\IsClosed]]</a> slot is <code>true</code>.
+                </td>
               </tr>
               <tr>
-                <td><dfn data-idl><code>connecting</code></dfn></td>
-                <td>Any of the <code><a>RTCIceTransport</a></code>s or
+                <td><dfn data-idl><code>failed</code></dfn></td>
+                <td>The previous state doesn't apply and any
+                <code><a>RTCIceTransport</a></code>s or
                 <code><a>RTCDtlsTransport</a></code>s are in the
-                <code>"connecting"</code> or <code>"checking"</code> state and none
-                of them is in the <code>"failed"</code> state.</td>
+                <code>"failed"</code> state.</td>
               </tr>
               <tr>
-                <td><dfn data-idl><code>connected</code></dfn></td>
-                <td>All <code><a>RTCIceTransport</a></code>s and
+                <td><dfn data-idl><code>disconnected</code></dfn></td>
+                <td>None of the previous states apply and any
+                <code><a>RTCIceTransport</a></code>s or
                 <code><a>RTCDtlsTransport</a></code>s are in the
-                <code>"connected"</code>, <code>"completed"</code> or
-                <code>"closed"</code> state and at least one of them is in the
-                <code>"connected"</code> or <code>"completed"</code> state.</td>
+                <code>"disconnected"</code> state.</td>
```

Comment:
Me thinking out loud:

So if you have two transports, one connected and one failed or disconnected, you would be failed or disconnected even if you do have a connection to the other side?

It sounds weird to be in "failed/disconnected" and still have data flowing to the other endpoint. But if "connecting" is prioritized over "connected" because SOMETHING is connecting, then it makes sense to say that we are "failed/disconnected" because SOMETHING is "failed/disconnected". Maybe it's better to highlight that "something is wrong" than to give a happy state - the caller can inspect individual transports for a full picture.

Otherwise we could have a "partially-connected" state, but I bet people don't want to add more enum values.

Reply:
Ok I think I am happy with this. :) I was thinking "what if one of the transports gets stuck in a 'bad' state?" but if that is the case then you want to know about it. And if a transport should be discarded, then we should make sure that the transport is thrown away rather than add hacks to make the enum produce something more helpful.